src/inference/Inference.jl

```diff
@@ -105,18 +105,28 @@ end
 # Default Transition #
 ######################
 
-struct Transition{T, F<:AbstractFloat}
-    θ  :: T
-    lp :: F
+struct Transition{T, F<:AbstractFloat, S<:Union{NamedTuple, Nothing}}
+    θ     :: T
+    lp    :: F
+    stat  :: S
```

Comment:
I wonder if we should simplify this to just
```suggestion
    stat  :: S
```
which would include `lp`. Then we also don't have to deal with the `nothing` case below and `metadata(t::Transition) = t.stat` (maybe that would motivate renaming the field even to `metadata`?).